wordpress-mobile/WordPress-iOS
# Fix/16961 - Events properties logic

Part of [#16961](https://github.com/wordpress-mobile/WordPress-iOS/issues/16961)

This PR replaces the logic for generating reader tracking properties (that is located in many places) with single point for calculation in `ReaderHelpers` class, more specifically `statsProperties(for post: mergeWith existing:)` method. Also, enums are used for properties keys, and only properties with meaningful values are added.

## To test
1. Trigger every event found in the events table found [here](https://github.com/wordpress-mobile/WordPress-Android/pull/14283). Trigger can be found in the Trigger column.
2. Make sure tracking event log in the console includes: `blog_id`, `feed_id` and `follow` properties.

## Regression Notes
1. Potential unintended areas of impact
n/a

2. What I did to test those areas of impact (or what existing automated tests I relied on)
n/a

3. What automated tests I added (or what prevented me from doing so)
n/a

PR submission checklist:
